### ISSUE TITLE
Review procedures for Secrets Backend

### ISSUE THREAD

**Issue by lzdanski** [astronomer/docs#2921]
### Issue Type

Inaccurate, misleading, or out-of-date information

### Links to Affected Docs

https://docs.astronomer.io/astro/secrets-backend

### The Issue or Context

There are some inconsistencies in structure, language, and style in the Secrets Backend configuration docs. They need to be reviewed and updated. 

### Fix or Proposed Change

- [ ] AWS Paramstore
- [ ] AWS Secrets Manager
- [ ] Hashicorp Vault
- [ ] Azure Key Store
- [ ] GCS Manager

### Additional Notes

Wait until after https://github.com/astronomer/docs/pull/2739 merges

**Comment by melugoyal** [comment]
couple suggestions as part of this:
- we can document the improvements made in https://github.com/apache/airflow/pull/34706, where instead of configuring credentials for Azure Key Vault as env vars, the credentials can be configured in the secrets backend kwargs. this will prevent conflicts with the same env vars being set for other use cases
- we can document that for authenticating with AWS Secrets Manager in Astro, people don't need to use static credentials and can instead set `role_arn` in `backend_kwargs`: https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/secrets-backends/aws-secrets-manager.html#aws-secrets-manager-backend

**Comment by melugoyal** [comment]
@lzdanski do you think we can address the above soon?

**Comment by lzdanski** [comment]
@melugoyal is that work related to this issue that was made a couple of weeks ago? https://github.com/astronomer/docs/issues/2916

**Comment by melugoyal** [comment]
yup the Azure Key Vault Secrets backend piece is related, specifically the part i mentioned about `this will prevent conflicts with the same env vars being set for other use cases` is what was being referenced in #2916

**Comment by melugoyal** [comment]
@lzdanski are we still revisiting these docs? i think in addition to my feedback above, having an example in the docs of how to leverage https://github.com/apache/airflow/pull/35052 to use Azure Key Vault as a secrets backend via workload identity and recommending it as a best practice would be great.

**Comment by lzdanski** [comment]
We need a dedicated technical partner to work with to make these updates, and more immediate new features have been taking priority for docs work.

Will reopen PRs later with dedicated partner.
